packages/extension-base/src/koni/background/handlers/State.ts

```diff
@@ -1177,6 +1182,345 @@ export default class KoniState {
+    }
+
+    if (!(network === 'mainnet' || network === 'testnet')) {
+      throw new BitcoinProviderError(BitcoinProviderErrorType.INVALID_PARAMS, t('Network to try this request is must be mainnet or testnet'));
+    }
+
+    if (!isBitcoinAddress(address)) {
+      throw new BitcoinProviderError(BitcoinProviderErrorType.INVALID_PARAMS, t('Not found address'));
+    }
+
+    // Check sign abiblity
+    if (!allowedAccounts.find((acc) => (acc.toLowerCase() === address.toLowerCase()))) {
+      throw new BitcoinProviderError(BitcoinProviderErrorType.INVALID_PARAMS, t('You have rescinded allowance for this account in wallet'));
+    }
+
+    const pair = keyring.getPair(address);
+
+    if (!pair) {
+      throw new BitcoinProviderError(BitcoinProviderErrorType.INVALID_PARAMS, t('Unable to find account'));
+    }
+
+    if (network === 'mainnet') {
+      if (!['bitcoin-86', 'bitcoin-84'].includes(pair.type)) {
+        throw new BitcoinProviderError(BitcoinProviderErrorType.INVALID_PARAMS, t('Your address is not on the mainnet network'));
+      }
+    } else if (network === 'testnet') {
+      if (!['bittest-86', 'bittest-84'].includes(pair.type)) {
+        throw new BitcoinProviderError(BitcoinProviderErrorType.INVALID_PARAMS, t('Your address is not on the testnet network'));
+      }
+    }
+
+    const account: AccountJson = { address: pair.address, ...pair.meta };
+
+    const psbtGenerate = bitcoin.Psbt.fromHex(psbt);
+    const psbtTxInputs = psbtGenerate.txInputs;
+    const psbtTxOutputs = psbtGenerate.txOutputs;
+
+    const payload: BitcoinSignPsbtPayload = {
+      psbt: psbtGenerate,
+      broadcast: !!broadcast,
+      network,
+      signAtIndex: isArray(signAtIndex) && signAtIndex.length === 0 ? undefined : signAtIndex,
+      account: account.address,
+      allowedSighash,
+      txInput: psbtTxInputs,
+      txOutput: psbtTxOutputs
+    };
+    const hashPayload = '';
+    const canSign = !account.isExternal;
+
+    const signPayload: BitcoinSignPsbtRequest = {
+      account,
+      payload,
+      hashPayload,
+      canSign
+    };
+
+    return this.requestService.addConfirmationBitcoin(id, url, 'bitcoinSignPsbtRequest', signPayload, {
+      requiredPassword: false
+    })
+      .then(({ isApproved, payload }) => {
+        if (isApproved) {
+          if (payload) {
+            return payload;
+          } else {
+            throw new BitcoinProviderError(BitcoinProviderErrorType.INVALID_PARAMS, t('Not found signature'));
+          }
+        } else {
+          throw new BitcoinProviderError(BitcoinProviderErrorType.USER_REJECTED_REQUEST);
+        }
+      });
+  }
+
+  public async bitcoinSendTransaction (id: string, url: string, networkKey: string, allowedAccounts: string[], transactionParams: BitcoinSendTransactionParams): Promise<string | undefined> {
+    const bitcoinApi = this.getBitcoinApi(networkKey);
```

Comment:
In type defined, `networkKey` is `mainnet` or `testnet`, but in apiMap, these keys do not existed